packages/react/src/components/table/DataTable.tsx

```diff
@@ -0,0 +1,266 @@
+import { useTheme } from '../../hooks/useTheme';
+
+type Header = {
+  key: string;
+  headerName: string;
+  sortIconType?: 'string' | 'other';
+  transform?: ({ args }: any) => string | JSX.Element; // eslint-disable-line @typescript-eslint/no-explicit-any
+};
+
+export interface TableCustomTheme {
+  '--background-color'?: string;
+}
+
+export type DataTableProps = React.ComponentPropsWithoutRef<'table'> & {
+  cellConfig: {
+    cols: Array<Header>;
+    sortingEnabled?: boolean;
+    initialSortingColumnKey?: string; // undefined -> neutral order for all columns
+    initialSortingOrder?: 'asc' | 'desc';
+    ariaLabelSortButtonNeutral?: string;
+    ariaLabelSortButtonAscending?: string;
+    ariaLabelSortButtonDescending?: string;
+    indexKey: string; // column key used as unique identifier for row
+    renderIndexCol?: boolean; // whether index colum is rendered in table. Defaults to true.
+  };
+  rows: Array<object>;
+  verticalHeaders?: Array<Header>;
+  variant?: 'dark' | 'light';
+  dense?: boolean;
+  zebra?: boolean;
+  caption?: string | React.ReactNode;
+  verticalLines?: boolean;
+  textAlignContentRight?: boolean; // defaults to false -> text is aligned left
+  checkboxSelection?: boolean;
+  ariaLabelCheckboxSelection?: string;
+  setSelections?: Function; // Callback that gets called with all selected row id values
+  initiallySelectedRows?: any[]; // Initially selected rows. Apply corresponding indexKey values here.
+  /**
+   * Custom theme styles
+   */
+  theme?: TableCustomTheme;
+};
+
+function processRows(rows, order, sorting, cellConfig) {
+  if (!cellConfig.sortingEnabled || !order || !sorting) {
+    return [...rows];
+  }
+
+  const sortColumn = cellConfig.cols.find((column) => {
+    return column.key === sorting;
+  });
+
+  const customSortCompareFunction = sortColumn ? sortColumn.customSortCompareFunction : undefined;
+
+  if (customSortCompareFunction) {
+    const sortedRows = [...rows].sort((a, b) => {
+      const aValue = a[sorting];
+      const bValue = b[sorting];
+
+      return customSortCompareFunction(aValue, bValue);
+    });
+
+    if (order === 'asc') {
+      return sortedRows;
+    }
+    if (order === 'desc') {
+      return sortedRows.reverse();
+    }
+  }
+
+  return [...rows].sort((a, b) => {
+    const aValue = a[sorting];
+    const bValue = b[sorting];
+
+    if (aValue < bValue) {
+      return order === 'asc' ? -1 : 1;
+    }
+    if (aValue > bValue) {
+      return order === 'asc' ? 1 : -1;
+    }
+
+    return 0;
+  });
+}
+
+export const DataTable = React.forwardRef(
+  (
+    {
+      cellConfig,
+      checkboxSelection,
+      ariaLabelCheckboxSelection,
+      initiallySelectedRows,
+      rows,
+      variant = 'dark',
+      dense = false,
+      zebra = false,
+      verticalLines = false,
+      verticalHeaders,
+      textAlignContentRight = false,
+      setSelections,
+      caption,
+      theme,
+      ...rest
+    }: DataTableProps,
+    ref?: any,
```

Comment:
Is this ref for table element?

Reply:
In the current implementation it's ref for the DataTable component. Only used inside the imperativeHandle hook:

`    useImperativeHandle(ref, () => ({
      selectAllRows() {
        selectAllRows();
      },
      deSelectAllRows() {
        deSelectAllRows();
      },
    }));`

Reply:
I guess the type could then be HtmlTableRef or something?

Reply:
True. I wonder if we should then also pass it to the html table element, as the name suggests?

Reply:
Probably 👍